### PR TITLE
Add example project for postgres

### PR DESCRIPTION
**Pull request by Prabeshpd**
Add example project that uses postgres.

### CODE REVIEW

File: examples/node-app-pg/connections.sync-db.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (kabirbaidhya) [inline examples/node-app-pg/connections.sync-db.json]: This file should not be committed. It should be in the gitignore.

File: examples/node-app-pg/knexfile.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+//     client: "postgresql",
+//     connection: {
+//       database: "my_db",
+//       user: "username",
+//       password: "password"
+//     },
+//     pool: {
+//       min: 2,
+//       max: 10
+//     },
```
Comment (kabirbaidhya) [inline examples/node-app-pg/knexfile.js]: remove all these commented lines.

File: examples/node-app-pg/knexfile.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+require("@babel/register");
+const { connections } = require("./connections.sync-db.json");
+module.exports = {
```
Comment (mesaugat) [inline examples/node-app-pg/knexfile.js]: New line after require.

File: examples/node-app-pg/src/migrations/20191202053138_users.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+exports.up = function(knex) {
```
Comment (mesaugat) [inline examples/node-app-pg/src/migrations/20191202053138_users.js]: Use the ES6 syntax here.

```js
export function up() {
}
```

File: examples/node-app-pg/src/migrations/20191202053138_users.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    table.string('password').notNullable();
+    table.string('firstname').notNullable();
+    table.string('lastname').notNullable();
+    table.string('username').notNullable();
+    table.timestamp('created_at').defaultTo(knex.fn.now());
+    table.timestamp('updated_at').defaultTo(knex.fn.now());
+  });
+};
+
+exports.down = function(knex) {
```
Comment (mesaugat) [inline examples/node-app-pg/src/migrations/20191202053138_users.js]:
```js
export function down() {
...
}
```

File: examples/node-app-pg/src/migrations/20191202053206_todos.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+exports.up = function(knex) {
```
Comment (mesaugat) [inline examples/node-app-pg/src/migrations/20191202053206_todos.js]: Same.

File: examples/node-app-pg/src/migrations/20191202053206_todos.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    table
+      .foreign("user")
+      .references("userId")
+      .inTable("users");
+    table.timestamp("created_at").defaultTo(knex.fn.now());
+    table.timestamp("updated_at").defaultTo(knex.fn.now());
+  });
+};
+
+exports.down = function(knex) {
```
Comment (mesaugat) [inline examples/node-app-pg/src/migrations/20191202053206_todos.js]: Same.

File: examples/node-app-pg/src/sql/function/dbo/sum.sql [hunk 8]
```diff
@@ -0,0 +1,5 @@
+CREATE FUNCTION [dbo].square(@x INT) RETURNS INT
+AS
+BEGIN
+    RETURN @x * @x;
+END
```
Comment (mesaugat) [inline examples/node-app-pg/src/sql/function/dbo/sum.sql]: @Prabeshpd You are missing a new line at the end of the file. Please enable it in your editor. 

Add this setting if you are using VSCode and then save all the files again.

```json
"files.insertFinalNewline": true
``

File: examples/node-app-pg/src/sql/procedure/utils/get_date.sql [hunk 9]
```diff
@@ -0,0 +1,8 @@
+--
+-- Procedure that select GETDATE function.
+--
+CREATE PROCEDURE [utils].get_date
+AS
+BEGIN
+    SELECT GETDATE() AS result;
+END
```
Comment (mesaugat) [inline examples/node-app-pg/src/sql/procedure/utils/get_date.sql]: Same.

Screen Shot 2019-12-19 at 12 44 00 PM

File: examples/node-app-pg/src/sql/schema/utils.sql [hunk 10]
```diff
@@ -0,0 +1 @@
+CREATE SCHEMA utils;
```
Comment (mesaugat) [inline examples/node-app-pg/src/sql/schema/utils.sql]: .

File: examples/node-app-pg/src/sql/view/utils/vw_table_names.sql [hunk 11]
```diff
@@ -0,0 +1,11 @@
+--
+-- View to fetch table names in the database.
+--
+CREATE VIEW [utils].vw_table_names
+AS
+    (
+    SELECT t.name
+    FROM sys.tables t
+);
+
```
Comment (mesaugat) [inline examples/node-app-pg/src/sql/view/utils/vw_table_names.sql]: Remove multiple new lines at the end. We only need a line feed.

File: examples/node-app-pg/.gitignore [hunk 12]
```diff
@@ -0,0 +1 @@
+./node_modules
```
Comment (mesaugat) [inline examples/node-app-pg/.gitignore]: .

File: examples/node-app-pg/.prettierignore [hunk 13]
```diff
@@ -0,0 +1 @@
+./src/sql
```
Comment (mesaugat) [inline examples/node-app-pg/.prettierignore]: .

File: examples/node-app-pg/Dockerfile [hunk 14]
```diff
@@ -0,0 +1,5 @@
+From node:10
+WORKDIR /app/
+COPY . .
+RUN yarn
+CMD yarn sync-db && yarn start
```
Comment (mesaugat) [inline examples/node-app-pg/Dockerfile]: Also add the `editorconfig` extension in VSCode.

File: examples/node-app-pg/src/index.js [hunk 15]
```diff
@@ -0,0 +1,40 @@
+const knex = require('knex');
+
+const { connections } = require('../connections.sync-db.json');
+
+const as = async () => {
```
Comment (mesaugat) [inline examples/node-app-pg/src/index.js]: No need to create a variable here. Just do this.

```js
(async () => {
...
})();
```

File: examples/node-app-pg/package.json [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+    "lint:fix": "eslint --fix src/index.js",
+    "sync-db": "sync-db",
+    "start": " babel-node src/index.js",
+    "make": "knex --knexfile=knexfile.js migrate:make",
+    "migrate": "knex --knexfile=knexfile.js migrate:latest",
+    "rollback": "knex --knexfile=knexfile.js migrate:rollback"
+  },
+  "license": "MIT",
+  "dependencies": {
+    "@leapfrogtechnology/sync-db": "^1.0.0-alpha.6",
```
Comment (kabirbaidhya) [inline examples/node-app-pg/package.json]: Use the latest version.
```suggestion
    "@leapfrogtechnology/sync-db": "^1.0.0-alpha.8",
```

File: examples/node-app-pg/package.json [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+    "eslint": "^6.7.2",
+    "eslint-config-leapfrog": "^2.0.1",
```
Comment (kabirbaidhya) [inline examples/node-app-pg/package.json]: these should be in dev dependencies.

**Comment by kabirbaidhya** [comment]
@Prabeshpd Few changes:
1. Use `yarn` instead of `npm` and remove `package-lock.json` file. You should commit `yarn.lock` file instead.

File: examples/node-app-pg/package.json [hunk 18]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "app-pg",
+  "description": "Sample project using sync-db",
+  "scripts": {
+    "lint": "eslint src/index.js",
+    "lint:fix": "eslint --fix src/index.js",
+    "sync-db": "sync-db",
+    "start": " babel-node src/index.js",
```
Comment (kabirbaidhya) [inline examples/node-app-pg/package.json]:
```suggestion
    "start": "babel-node src/index.js",
```

File: examples/node-app-pg/src/index.js [hunk 19]
```diff
@@ -1,40 +1,44 @@
...
 
-const { connections } = require('../connections.sync-db.json');
-
-const as = async () => {
+(async () => {
   try {
     const db = knex({
-      client: 'postgresql',
-      connection: connections.find(({ id }) => id === 'db1')
+      client: "postgresql",
```
Comment (mesaugat) [inline examples/node-app-pg/src/index.js]: Single quotes.

File: examples/node-app-pg/src/migrations/20191202053138_users.js [hunk 20]
```diff
@@ -1,16 +1,28 @@
...
-    table.timestamp('created_at').defaultTo(knex.fn.now());
-    table.timestamp('updated_at').defaultTo(knex.fn.now());
+/**
+ * Create table `users`.
+ *
+ * @param {Knex} knex
+ * @returns {Promise<any>}
+ */
+exports.up = knex => {
+  return knex.schema.createTable("users", table => {
```
Comment (mesaugat) [inline examples/node-app-pg/src/migrations/20191202053138_users.js]: single quotes.

File: examples/node-app-pg/src/migrations/20191202053206_todos.js [hunk 21]
```diff
@@ -1,4 +1,10 @@
-exports.up = function(knex) {
+/**
+ * Create table `todos`.
+ *
+ * @param {Knex} knex
+ * @returns {Promise<any>}
+ */
+exports.up = knex => {
   return knex.schema.createTable("todos", table => {
```
Comment (mesaugat) [inline examples/node-app-pg/src/migrations/20191202053206_todos.js]: Let's use single quotes.

Comment (kabirbaidhya) [reply examples/node-app-pg/src/migrations/20191202053206_todos.js]: `yarn lint:fix`

File: examples/node-app-pg/.prettierignore (hunk 13 above)
Comment (kabirbaidhya) [inline examples/node-app-pg/.prettierignore]: ?

File: examples/node-app-pg/connections.sync-db.json.example [hunk 22]
```diff
@@ -0,0 +1,12 @@
+{
+  "connections": [
+    {
+      "id": "db1",
+      "host": "postgresdb",
+      "user": "",
```
Comment (kabirbaidhya) [inline examples/node-app-pg/connections.sync-db.json.example]: testdb

File: examples/node-app-pg/connections.sync-db.json.example [hunk 23]
```diff
@@ -0,0 +1,12 @@
...
+      "port": 5432,
+      "database": "mydatabasename",
```
Comment (kabirbaidhya) [inline examples/node-app-pg/connections.sync-db.json.example]: testdb

File: examples/node-app-pg/docker-compose.yml [hunk 24]
```diff
@@ -0,0 +1,13 @@
+version: "3.3"
+services:
+  db:
+    image: "postgres"
+    container_name: "example-pg"
+    ports:
+      - "5432:5432"
+    environment:
+      DATABASE_URL: ${connectionString}
```
Comment (kabirbaidhya) [inline examples/node-app-pg/docker-compose.yml]:
```suggestion
      DATABASE_URL: ${DATABASE_URL}
```

File: examples/node-app-pg/src/migrations/20191202053138_users.js [hunk 25]
```diff
@@ -0,0 +1,28 @@
...
+    table.increments("userId");
```
Comment (kabirbaidhya) [inline examples/node-app-pg/src/migrations/20191202053138_users.js]: id

File: examples/node-app-pg/src/migrations/20191202053206_todos.js [hunk 26]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Create table `todos`.
+ *
+ * @param {Knex} knex
+ * @returns {Promise<any>}
+ */
+exports.up = knex => {
+  return knex.schema.createTable("todos", table => {
+    table.increments("tableId");
```
Comment (kabirbaidhya) [inline examples/node-app-pg/src/migrations/20191202053206_todos.js]: id

File: examples/node-app-pg/src/migrations/20191202053206_todos.js [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+    table.string("name").notNullable();
+    table.enu("status", ["active", "inactive"]).notNullable();
+    table
+      .foreign("user")
+      .references("userId")
```
Comment (kabirbaidhya) [inline examples/node-app-pg/src/migrations/20191202053206_todos.js]: id

File: examples/node-app-pg/src/sql/function/dbo/square.sql [hunk 28]
```diff
@@ -0,0 +1,9 @@
+--
+-- Calculate square of a number.
+--
+CREATE FUNCTION [dbo].square(@x INT) RETURNS INT
+AS
+BEGIN
+    RETURN @x * @x;
```
Comment (kabirbaidhya) [inline examples/node-app-pg/src/sql/function/dbo/square.sql]:
```suggestion
  RETURN @x * @x;
```

File: examples/node-app-pg/src/sql/function/dbo/sum.sql [hunk 29]
```diff
@@ -0,0 +1,8 @@
+--
+-- Calculate square of a number.
+--
+CREATE FUNCTION [dbo].square(@x INT) RETURNS INT
+AS
+BEGIN
+    RETURN @x * @x;
```
Comment (kabirbaidhya) [inline examples/node-app-pg/src/sql/function/dbo/sum.sql]:
```suggestion
  RETURN @x * @x;
```

File: examples/node-app-pg/src/sql/function/utils/product.sql [hunk 30]
```diff
@@ -0,0 +1,8 @@
+--
+-- Calculate multiplication of two numbers.
+--
+CREATE FUNCTION [utils].product(@a INT, @b INT) RETURNS INT
+AS
+BEGIN
+    RETURN @a * @b;
```
Comment (kabirbaidhya) [inline examples/node-app-pg/src/sql/function/utils/product.sql]:
```suggestion
  RETURN @a * @b;
```

**Comment by kabirbaidhya** [comment]
@Prabeshpd Delete `package-lock.json` file. We use yarn.
image

File: README.md [hunk 31]
```diff
@@ -172,6 +172,7 @@ await db.transaction(async trx => {
 
 1. [Node MSSQL Sample](examples/node-app-mssql)
 2. [Node MSSQL Programmatic Usage Sample](examples/node-mssql-programmatic-use)
+3. [Node POSTGRESQL Sample](examples/node-app-pg)
```
Comment (kabirbaidhya) [inline README.md]:
```suggestion
3. [Node PostgreSQL Sample](examples/node-app-pg)
```

**Pull request merged**